### ISSUE TITLE
Simplify CountryHistoryMixGraph data processing based on displayByEmissions value

### ISSUE THREAD

**Issue by fbarl** [tmrowco/electricitymap-contrib#2312]
That graph actually shows very different content depending on whether the _Carbon emissions_ tab is selected in the country panel or not.

Splitting it into two separate graph wrappers and branching into one of them depending on `tableDisplayEmissions` would make sense semantically and would also make the code simpler.

**Comment by corradio** [comment]
Would it be possible to pass a prop "displayByEmissions" to <CountryHistoryCarbonGraph> in order to selectively use a different `prepareGraphData` and re-use the underlying <AreaGraph> instance?
That would enable animations etc.. and avoid hiding/reconstructing a new graph everytime.

**Comment by fbarl** [comment]
> Would it be possible to pass a prop "displayByEmissions" to CountryHistoryCarbonGraph in order to selectively use a different `prepareGraphData` ...

Yes, that's definitely what I had in mind as a part of the solution.

>  and re-use the underlying AreaGraph instance? That would enable animations etc.. and avoid hiding/reconstructing a new graph everytime.

I still have my doubts whether animations in this case are a good idea for the same reason as I argued in https://github.com/tmrowco/electricitymap-contrib/pull/2313#discussion_r398001302, but if you feel strongly that they would improve the user experience, we can add an issue for re-introducing them to the `AreaGraph` component and keep this case in mind.

The graph initialization and rendering is still very cheap, actually - e.g. with the current carbon / emissions graph, I get a very smooth transition when switching between the tabs even though a graph gets destroyed and rendered fresh since #2313.

**Comment by corradio** [comment]
Upon thinking a bit more about it, I think it makes sense to allow for re-using the same underlying AreaGraph for some cases.
For example, when switching between countries, it makes sense that the bars & lines shrink/grow.
Furthermore, when switching to the emission tabs, it also makes sense that the bars & lines shrink/grow as it shows how much a % of the mix relates to a % of emissions.
I think we could do everything as part of this PR?

**Comment by fbarl** [comment]
> Upon thinking a bit more about it, I think it makes sense to allow for re-using the same underlying AreaGraph for some cases. For example, when switching between countries, it makes sense that the bars & lines shrink/grow.

You are right, that is indeed a good use case! :+1:

> Furthermore, when switching to the emission tabs, it also makes sense that the bars & lines shrink/grow as it shows how much a % of the mix relates to a % of emissions.

Ah, what I was concerned about were the graph / axis units (the percentages only appear in tooltips as inferred values). However, in this particular case the mix graph usually doesn't look as wildly different between consumption and emissions as I thought and since the two are highly correlated, I'm on board with making them animated - I tried it out and it does look good :smile: 

So in summary, I created #2336 to add animations separately to keep this issue just about how we organize and pass the graph data. I'll update the title to reflect that.

**Comment by fbarl** [comment]
De-assigning myself for now as this doesn't seem like a top priority.

**Comment by corradio** [comment]
Closing due to inactivity
